stac_fastapi/extensions/setup.py

```diff
@@ -7,8 +7,8 @@
 
 install_requires = [
     "attrs",
-    "pydantic[dotenv]<2",
-    "stac_pydantic==2.0.*",
+    "pydantic[dotenv]>=2",
+    "stac_pydantic>=3",
```

Comment:
same for `api` 
- `dotenv` is not an option anymore
- limit `stac_pydantic` upper version
- pin `stac-fastapi.types` and `stac-fastapi.api` version